packages/react-router/__tests__/setup.ts

```diff
@@ -1,15 +1,19 @@
 // https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html#configuring-your-testing-environment
 globalThis.IS_REACT_ACT_ENVIRONMENT = true;
 
-if (!globalThis.fetch) {
+if (!globalThis.TextEncoder || !globalThis.TextDecoder) {
   const { TextDecoder, TextEncoder } = require("node:util");
-  globalThis.TextDecoder = TextDecoder;
   globalThis.TextEncoder = TextEncoder;
+  globalThis.TextDecoder = TextDecoder;
+}
 
+if (!globalThis.ReadableStream || !globalThis.WritableStream) {
   const { ReadableStream, WritableStream } = require("node:stream/web");
   globalThis.ReadableStream = ReadableStream;
   globalThis.WritableStream = WritableStream;
+}
 
+if (!globalThis.fetch) {
```

Comment:
I'm not quite sure why this is still needed - but unit tests fail without it - maybe a `jsdom` issue?

```
>  pnpm test packages/react-router/__tests__/router/navigation-test.ts
...
  ● navigations › navigation states › loader submission + redirect

    ReferenceError: Request is not defined

      4978 |   }
      4979 |
    > 4980 |   return new Request(url, init);
           |   ^
      4981 | }
      4982 |
      4983 | function convertFormDataToSearchParams(formData: FormData): URLSearchParams {

      at createClientSideRequest (lib/router/router.ts:4980:3)
      at createClientSideRequest (lib/router/router.ts:1560:19)
      at Object.startNavigation [as navigate] (lib/router/router.ts:1411:11)
      at Object.navigate (__tests__/router/utils/data-router-setup.ts:642:19)
      at Object.navigate (__tests__/router/navigation-test.ts:1250:23)
```